src/client/components/vision2/camera/world_line.ts

```diff
@@ -0,0 +1,136 @@
+import * as THREE from 'three'
+
+import { Vector2 } from '../../../math/vector2'
+import { Vector3 } from '../../../math/vector3'
+import { Vector4 } from '../../../math/vector4'
+import { mesh } from '../../three/builders'
+import { planeGeometry } from '../../three/builders'
+import { rawShader } from '../../three/builders'
+import { shaderMaterial } from '../../three/builders'
+import { createUpdatableComputed } from '../../three/create_updatable_computed'
+import { Canvas } from '../../three/three'
+
+import { Lens } from './model'
+import fragmentShader from './shaders/world_line.frag'
+import vertexShader from './shaders/world_line.vert'
+
+/** If start === end this will draw the entire cone. */
+export interface ConeSegment {
+  /** The normal axis of the plane. This is only needed if start and end are parallel or anti-parallel. */
+  axis?: Vector3,
+  /** The vector pointing to the start of the line segment. */
+  start: Vector3,
+  /** The vector pointing to the end of the line segment. */
+  end: Vector3,
+  /** The colour of the line to draw. */
+  color?: Vector4,
+  /** The width of the line to draw on the screen in pixels. */
+  lineWidth?: number
+}
+
+export class WorldLine {
+  constructor(
+    private readonly canvas: Canvas,
+    private readonly lens: () => Lens,
+  ) {
+  }
+
+  static of(canvas: Canvas, lens: () => Lens) {
+    return new WorldLine(canvas, lens)
+  }
+
+  /** Draws a segment of a plane projected to infinity in world space. */
```

Comment:
Why does plane segment get a comment, but not the others?

Reply:
Should all of these functions get docstrings? I have noticed that most of NUsight2 doesn't have them